tests/str/StrTransformTest.php

```diff
@@ -386,6 +434,136 @@ public function testReplaceEveryCI(
     expect(Str\replace_every_ci($haystack, $replacements))->toEqual($expected);
   }
 
+  public static function provideReplaceEveryNonrecursive(): varray<mixed> {
+    return varray[
+      tuple(
+        'hello world',
+        dict[
+          'h' => 'H',
+          'w' => 'W',
+        ],
+        'Hello World',
+      ),
+      tuple(
+        'Hello world',
+        dict[
+          'h' => 'H',
+          'H' => 'h',
+          'w' => 'W',
+          'W' => 'w',
+        ],
+        'hello World',
+      ),
+      tuple(
+        'hello world',
+        dict[
+          'hello' => 'hi',
+          'world' => 'universe',
+        ],
+        'hi universe',
+      ),
+      tuple(
+        'Hi hello world hello hi',
+        dict[
+          'hello' => 'hi',
+          'hi' => 'Hello',
+        ],
+        'Hi hi world hi Hello',
+      ),
+    ];
+  }
+
+  <<DataProvider('provideReplaceEveryNonrecursive')>>
+  public function testReplaceEveryNonrecursive(
+    string $haystack,
+    KeyedContainer<string, string> $replacements,
+    string $expected,
+  ): void {
+    expect(Str\replace_every_nonrecursive($haystack, $replacements))->toBeSame($expected);
+  }
+
+  public function testReplaceEveryNonrecursiveExceptions(): void {
+    expect(() ==> Str\replace_every_nonrecursive('hello world', dict['h' => 'H', '' => 'W']))
+      ->toThrow(InvariantException::class);
+  }
+
+  public static function provideReplaceEveryNonrecursiveCI(): dict<string, (string, dict<string, string>, string)> {
+    return dict[
+      'Replace lowercase letter with uppercase' => tuple(
+        'Hello world',
+        dict[
+          'h' => 'H',
+          'w' => 'W',
+        ],
+        'Hello World',
+      ),
+      'Replace multi character strings' => tuple(
+        'Hello world',
+        dict[
+          'Hello' => 'hi',
+          'World' => 'universe',
+        ],
+        'hi universe',
+      ),
+      'Replace both upper and lowercase strings with differing cases' => tuple(
+        'Hi hello world HELLO HI',
+        dict[
+          'HELLO' => 'hi',
+          'HI' => 'Hello',
+        ],
+        'Hello hi world hi Hello',
+      ),
+      'When a replacer is a prefix of another, the longer one wins' => tuple(
+        '12345',
+        dict[
+          '12' => 'FAIL',
+          '1234' => 'SUCCESS',
+        ],
+        'SUCCESS5',
+      ),
+      "Replacers do not look at the replacements from previous iterations" => tuple(
+        'sub fast',
+        dict[
+          'sub' => 'subject',
+          'bject' => 'FAIL',
```

Comment:
You could add more keys here to make the test more thorough (I can imagine a buggy implementation could fail on only some of them):

```
'bject' => 'FAIL',
'ject' => 'FAIL',
'c' => 'FAIL',
```